modules/Microsoft.DBforMySQL/flexibleServers/.test/public/deploy.test.bicep

```diff
@@ -0,0 +1,165 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.dbformysql.flexibleservers-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'dfmsfsp'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+@description('Generated. Used as a basis for unique resource names.')
+param baseTime string = utcNow('u')
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies1 'dependencies1.bicep' = {
```

Comment:
Why is this split in 2? I can see that an output of the 1st dependency deployment is passed to the 2nd, but I fail do unterstand why :D

Reply:
The `location` property cannot accept an output from a previous deployment. That's why I had to split it into two, so that I can extract the paired region via the deployment script, then pass it as an input to the next dependency deployment which is in the paired region 

```bicep
This expression is being used in an assignment to the "location" property of the "Microsoft.KeyVault/vaults" type, which requires a value that can be calculated at the start of the deployment. Properties of geoBackupManagedIdentity which can be calculated at the start include "apiVersion", "id", "name", "type". Bicep(BCP120)
```

Reply:
Can we resolve this @AlexanderSehr ?